src/db/types/dependencies.rs

```diff
@@ -0,0 +1,127 @@
+use crate::utils::Dependency;
+use derive_more::Deref;
+use semver::VersionReq;
+use serde::{Deserialize, Serialize};
+
+const DEFAULT_KIND: &str = "normal";
+
+/// The three possible representations of a dependency in our internal JSON format
+/// in the `releases.dependencies` column.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+enum Dep {
+    Two((String, VersionReq)),
```

Comment:
Isn't it possible to have variant names that make more sense? ^^'

Reply:
> Isn't it possible to have variant names that make more sense? ^^'

What do you imagine? 

The variants represent the differnt ways it's represented in the DB in JSON: 

1. `["vec_map","^0.0.1"]`
2. `["vec_map","^0.0.1","normal" ]`
3. `["vec_map, "^0.0.1","normal", false]`

Reply:
Well, the first one is the crate name, the second the version selector, the third and the fourth are mysterious. ^^'

Reply:
Wouldn't be much better, in here I think a struct would be better considering all n+1 variants share the same first fields as n.